packages/expo-updates/android/src/main/java/expo/modules/updates/manifest/BareManifest.kt

```diff
@@ -0,0 +1,97 @@
+package expo.modules.updates.manifest
+
+import android.util.Log
+import expo.modules.updates.UpdatesConfiguration
+import expo.modules.updates.UpdatesUtils
+import expo.modules.updates.db.entity.AssetEntity
+import expo.modules.updates.db.entity.UpdateEntity
+import expo.modules.updates.db.enums.UpdateStatus
+import expo.modules.updates.loader.EmbeddedLoader
+import org.json.JSONArray
+import org.json.JSONException
+import org.json.JSONObject
+import java.util.*
+
+class BareManifest private constructor(override val rawManifestJson: JSONObject,
+                                       private val mId: UUID,
+                                       private val mScopeKey: String,
+                                       private val mCommitTime: Date,
+                                       private val mRuntimeVersion: String,
+                                       private val mMetadata: JSONObject?,
+                                       private val mAssets: JSONArray?) : Manifest {
+    override val serverDefinedHeaders: JSONObject?
+        get() = null
+
+    override val manifestFilters: JSONObject?
+        get() = null
+
+    override val updateEntity: UpdateEntity
+        get() = UpdateEntity(mId, mCommitTime, mRuntimeVersion, mScopeKey).apply {
+            if (mMetadata != null) {
+                metadata = mMetadata
+            }
+            status = UpdateStatus.EMBEDDED
+        }
+
+    override val assetEntityList: List<AssetEntity>
```

Comment:
I lean towards making this a getters method instead of a property -- property accessors tend to imply loading a member variable: few or no side effects, expensive/time-consuming operations, and `x.y == x.y`. We may want to do the same for `updateEntity` and other fields as well -- it seems confusing if `x.y` returns a new object every time.

Reply:
For this particular example, I think the best thing is to delegate to [lazy](https://kotlinlang.org/docs/delegated-properties.html#lazy-properties) which does memoization for us if all dependencies of this property truly are immutable (I think these objects are immutable and can maybe be data objects as well).

I don't feel too strongly about properties vs functions in general (other PRs in this stack use functions for things that can be properties. Reading the internet seems to indicate that the general guideline is to use methods for things that aren't state. In this case, this seems like readonly state that can be memoized so a lazy property is probably the way to go.

Reply:
Getter methods often have both behavior and state. I don't think it's either/or. I mostly want to be cautious about future implementations of this method having effects that a programmer today would not anticipate. There's no real burden on callers writing out `manfest.getAssetEntityList()` so we should choose between a method and property based on whichever communicates what's going on better. (In contrast, having methods for simple property accesses would communicate that there may be side effects or extra computation when in fact there are none.)